Source/WebKit/UIProcess/mac/WebViewImpl.mm

```diff
@@ -4079,6 +4080,64 @@ static NSDragOperation kit(std::optional<WebCore::DragOperation> dragOperation)
     return true;
 }
 
+static void performDragWithLegacyFiles(RefPtr<WebPageProxy> page, Box<Vector<String>>&& fileNames, Box<WebCore::DragData>&& dragData, const String& pasteboardName)
+{
+    auto* networkProcess = page->websiteDataStore().networkProcessIfExists();
+    if (!networkProcess)
+        return;
+    networkProcess->sendWithAsyncReply(Messages::NetworkProcess::AllowFilesAccessFromWebProcess(page->protectedLegacyMainFrameProcess()->coreProcessIdentifier(), *fileNames), [page = WTFMove(page), fileNames, dragData, pasteboardName]() mutable {
+        if (!page)
+            return;
+        ensureOnMainRunLoop([page = WTFMove(page), fileNames, dragData, pasteboardName]() mutable {
```

Comment:
I'm pretty sure this is unnecessary.  sendWithAsyncReply should call the lambda on the main run loop.